web/console/src/app/views/Match/MatchScore/index.tsx

```diff
@@ -70,13 +95,105 @@ export const MatchScore: React.FC = () => {
+    const addCasperWallet = () => {
+        try {
+            const currentQueueClient = getCurrentQueueClient();
+
+            setQueueClient(currentQueueClient);
+
+            queueCasperActionAllowAddress(user.casperWallet, user.walletType, squad.id);
+        }
+        catch (error: any) {
+            toast.error('Something went wrong', {
+                position: toast.POSITION.TOP_RIGHT,
+                theme: 'colored',
+            });
+        }
+    };
+
+    /** Adds velas wallet address for earning reward. */
+    const addVelasWallet = async() => {};
+
+    /** Adds wallets addresses for earning reward. */
+    const addWallet = async() => {
+        try {
+            const addingWallets = new Map();
+
+            const addingWalletsTypes = [
+                {
+                    walletType: VELAS_WALLET_TYPE,
+                    mint: addVelasWallet,
+                },
+                {
+                    walletType: CASPER_WALLET_TYPE,
+                    mint: addCasperWallet,
+                },
+                {
+                    walletType: METAMASK_WALLET_TYPE,
+                    mint: addMetamaskWallet,
+                },
+            ];
+
+            addingWalletsTypes.forEach(addingWalletsType =>
+                addingWallets.set(addingWalletsType.walletType, addingWalletsType.mint));
+
+            await addingWallets.get(user.walletType)();
+        } catch (e) {
+            toast.error('Invalid transaction', {
+                position: toast.POSITION.TOP_RIGHT,
+                theme: 'colored',
+            });
+        }
+    };
+
+    /** Mints token with casper wallet. */
+    const casperMint = async(messageEvent: any) => {
+        const casperTransactionService = new CasperTransactionService(user.casperWallet);
+
+        await casperTransactionService.mintUDT(messageEvent.message.casperTransaction, messageEvent.message.rpcNodeAddress);
+    };
+
+    /** Mints token with metamask wallet. */
+    const metamaskMint = async(messageEvent: any) => {
+        await service.mintUDT(messageEvent.message.transaction);
+    };
+
+    /** Mints token with velas wallet. */
+    const velasMint = () => {};
+
     if (queueClient) {
-        queueClient.ws.onmessage = ({ data }: MessageEvent) => {
+        queueClient.ws.onmessage = async({ data }: MessageEvent) => {
             const messageEvent = JSON.parse(data);
-            service.mintUDT(messageEvent.message.transaction);
+
+            const mintingTokens = new Map();
+
+            const mintingTokensTypes = [
+                {
+                    walletType: VELAS_WALLET_TYPE,
+                    mint: velasMint,
+                },
+                {
+                    walletType: CASPER_WALLET_TYPE,
+                    mint: casperMint,
+                },
+                {
+                    walletType: METAMASK_WALLET_TYPE,
+                    mint: metamaskMint,
+                },
+            ];
```

Comment:
123-136 - duplicate logic

Reply:
I will create service for this in another pr